src/sentry/static/sentry/app/views/settings/projectSourceMaps/detail/index.tsx

```diff
@@ -0,0 +1,210 @@
+    const {location, router} = this.props;
+
+    router.push({
+      ...location,
+      query: {...location.query, cursor: undefined, query},
+    });
+  };
+
+  handleDelete = async (id: string) => {
+    addLoadingMessage(t('Removing file...'));
+
+    try {
+      await this.api.requestPromise(`${this.getFilesUrl()}${id}/`, {
+        method: 'DELETE',
+      });
+      // We might want to refetch the files here to not mess up pagination
+      this.setState(state => ({files: state.files.filter(f => f.id !== id)}));
+      addSuccessMessage(t('File removed.'));
+    } catch {
+      addErrorMessage(t('Unable to remove file. Please try again.'));
+    }
+  };
+
+  getQuery() {
+    const {query} = this.props.location.query;
+
+    return typeof query === 'string' ? query : undefined;
+  }
+
+  getEmptyMessage() {
+    if (this.getQuery()) {
+      return t('There are no source maps that match your search.');
+    }
+
+    return t('There are no source maps for this release.');
+  }
+
+  renderLoading() {
+    return this.renderBody();
+  }
+
+  renderFiles() {
+    const {files} = this.state;
+    const fileApiUrl = this.api.baseUrl + this.getFilesUrl();
+
+    if (!files.length) {
+      return null;
+    }
+
+    return files.map(file => {
+      return (
+        <SourceMapsFileRow
+          key={file.id}
+          file={file}
+          onDelete={this.handleDelete}
+          downloadUrl={`${fileApiUrl}${file.id}/?download=1`}
+        />
+      );
+    });
+  }
+
+  renderBody() {
+    const {loading, files, filesPageLinks} = this.state;
+    const {version, orgId, projectId} = this.props.params;
+    const {project} = this.props;
+
+    return (
+      <React.Fragment>
+        <SettingsPageHeader
+          title={`${t('Source Maps in')} ${formatVersion(version)}`}
+          action={
+            <ButtonBar gap={1}>
+              <Button
+                size="small"
+                to={`/settings/${orgId}/projects/${projectId}/source-maps/`}
+                icon={<IconChevron size="xs" direction="left" />}
+              >
+                {t('All Source Maps')}
+              </Button>
+              <Button
+                size="small"
+                to={`/organizations/${orgId}/releases/${encodeURIComponent(
+                  version
+                )}/?project=${project.id}`}
+                icon={<IconReleases size="xs" />}
+              >
+                {t('View Release')}
+              </Button>
+            </ButtonBar>
+          }
+        />
+
+        <Wrapper>
+          <TextBlock noMargin>{t('Uploaded source maps')}:</TextBlock>
+          <SearchBar
+            placeholder={t('Search source maps')}
+            onSearch={this.handleSearch}
+            query={this.getQuery()}
+          />
+        </Wrapper>
+
+        <StyledPanelTable
+          headers={[
+            t('Name'),
+            t('Size'),
+            <TextRight key="actions">{t('Actions')}</TextRight>,
```

Comment:
minor nit: generally, I would try to name it for its contents, rather than its visual function

Reply:
Makes sense 👍